pkg/sql/colexec/auto_incr.go

```diff
@@ -75,28 +81,39 @@ func UpdateInsertValueBatch(e engine.Engine, ctx context.Context, proc *process.
 		return err
 	}
 	rel.Ranges(ctx, nil) // TODO
-	return UpdateInsertBatch(e, db, ctx, proc, ColDefs, bat, rel.GetTableID(ctx))
+	return UpdateInsertBatch(e, db, proc, ColDefs, bat, rel.GetTableID(ctx), dbName, tblName)
 }
 
 func getRangeFromAutoIncrTable(param *AutoIncrParam, bat *batch.Batch, tableID string) ([]uint64, []uint64, error) {
-	offset, step := make([]uint64, 0), make([]uint64, 0)
 	var err error
+	param.proc.TxnOperator, err = NewTxn(param.eg, param.proc, param.ctx)
```

Comment:
This is wrong, this forces an override of the old transaction context.

Reply:
@nnsgmsone WHY this override old transaction context?   How would people know?   It is because you passed in param.ctx or the return value overrite param.proc.TxnOperator?     Does this mean our NewTxn API is under documented?

Reply:
proc.TxnOperator should actually be removed. I'm not sure exactly who added this, but now auto_incr directly overrides this is causing some problems. I'm looking forward to removing all proc. @fengttt

Reply:
I modify this code.